src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ExtensionMethods;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    /// <summary>
+    /// Tracer integration for HttpMessageHandler.
+    /// </summary>
+    public static class HttpMessageHandlerIntegration
+    {
+        internal const string OperationName = "http.request";
+
+        // internal readonly  string Name = nameof(HttpMessageHandlerIntegration).Substring(0, )
+
+        /// <summary>
+        /// Instrumentation wrapper for <see cref="HttpMessageHandler.SendAsync"/>.
+        /// </summary>
+        /// <param name="handler">The <see cref="HttpMessageHandler"/> instance to instrument.</param>
+        /// <param name="request">The <see cref="HttpRequestMessage"/> that represents the current HTTP request.</param>
+        /// <param name="cancellationTokenSource">The <see cref="CancellationTokenSource"/> that can be used to cancel this <c>async</c> operation.</param>
+        /// <returns>Returns the value returned by the inner method call.</returns>
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Http",
+            TargetType = "System.Net.Http.HttpMessageHandler")]
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Http",
+            TargetType = "System.Net.Http.HttpClientHandler")]
+        public static object SendAsync(
+            object handler,
+            object request,
+            object cancellationTokenSource)
+        {
+            // HttpMessageHandler
+            // Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            var tokenSource = cancellationTokenSource as CancellationTokenSource;
+            var cancellationToken = tokenSource?.Token ?? CancellationToken.None;
+
+            return SendAsyncInternal(
+                (HttpMessageHandler)handler,
+                (HttpRequestMessage)request,
+                cancellationToken);
+        }
+
+        private static async Task<HttpResponseMessage> SendAsyncInternal(
+            HttpMessageHandler handler,
+            HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            var executeAsync = DynamicMethodBuilder<Func<HttpMessageHandler, HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>>>
+               .GetOrCreateMethodCallDelegate(
+                    handler.GetType(),
+                    nameof(SendAsync));
+
+            if (!IsTracingEnabled(request))
+            {
+                return await executeAsync(handler, request, cancellationToken);
```

Comment:
Thanks! I've made this mistake a few times already, so I just installed a R# extension that will warn me when I miss them 😉

Reply:
Setting up a Roslyn analyser such as https://www.nuget.org/packages/ConfigureAwaitChecker.Analyzer would be more robust since everyone would benefit from it. It seems that it's as simple as adding the nuget dependency to the project.

Reply:
Good idea. I did some research and found that there are newer (pre-release) versions of StyleCop that support newer C# 7.0+ syntax. The FxCop analyzers also include the check for `ConfigureAwait()` (among lots of other checks), so we should look into adding that one. I created a Trello card to follow-up on this later.